src/AdvApi32.Desktop/AdvApi32.Helpers.cs

```diff
@@ -0,0 +1,104 @@
+    /// a slightly higher level of functionality to ease calling into native code.
+    /// </content>
+    public static partial class AdvApi32
+    {
+        public static void InstallService(string servicePath, string serviceName, string serviceDisplayName, string serviceDescription, string userName, string password)
+        {
+            IntPtr hSCManager = OpenSCManager(null, null, ServiceManagerAccess.SC_MANAGER_CREATE_SERVICE);
+
+            if (hSCManager == IntPtr.Zero)
+            {
+                throw new Win32Exception(Marshal.GetHRForLastWin32Error());
+            }
+
+            IntPtr svcHandle = CreateService(
+                hSCManager,
+                serviceName,
+                serviceDisplayName,
+                ServiceAccess.SERVICE_ALL_ACCESS,
+                ServiceType.SERVICE_WIN32_OWN_PROCESS,
+                ServiceStartType.SERVICE_DEMAND_START,
+                ServiceErrorControl.SERVICE_ERROR_NORMAL,
+                servicePath,
+                null,
+                0,
+                null,
+                userName,
+                password);
+
+            if (svcHandle == IntPtr.Zero)
+            {
+                throw new Win32Exception(Marshal.GetHRForLastWin32Error());
+            }
+
+            ServiceDescription descriptionStruct = new ServiceDescription
+            {
+                lpDescription = serviceDescription
+            };
+
+            IntPtr lpInfo = Marshal.AllocHGlobal(Marshal.SizeOf(descriptionStruct));
+
+            if (lpInfo == IntPtr.Zero)
+            {
+                throw new Win32Exception(Marshal.GetHRForLastWin32Error());
+            }
+
+            Marshal.StructureToPtr(descriptionStruct, lpInfo, false);
+
+            if (!ChangeServiceConfig2(svcHandle, ServiceInfoLevel.SERVICE_CONFIG_DESCRIPTION, lpInfo))
+            {
+                Marshal.FreeHGlobal(lpInfo);
+                throw new Win32Exception(Marshal.GetHRForLastWin32Error());
+            }
+
+            Marshal.FreeHGlobal(lpInfo);
+
+            if (svcHandle == IntPtr.Zero)
+            {
+                throw new Win32Exception(Marshal.GetHRForLastWin32Error());
+            }
+
+            if (CloseServiceHandle(svcHandle) == IntPtr.Zero)
+            {
+                throw new Win32Exception(Marshal.GetHRForLastWin32Error());
+            }
+        }
+
+        public static void UninstallService(string serviceName)
+        {
+            IntPtr scmHandle = OpenSCManager(null, null, ServiceManagerAccess.GenericWrite);
+
+            if (scmHandle == IntPtr.Zero)
+            {
+                throw new Win32Exception(Marshal.GetHRForLastWin32Error());
+            }
+
+            IntPtr svcHandle = OpenService(scmHandle, serviceName, ServiceAccess.Delete);
+            if (svcHandle == IntPtr.Zero)
+            {
+                throw new Win32Exception(Marshal.GetHRForLastWin32Error());
```

Comment:
If there is an error here, the SCManager handle is never closed. Having a SafeHandle for it would allow to use `using` and let the `IDisposable` implementation do the cleaning.